### ISSUE TITLE
Problems with the DME

### ISSUE THREAD

**Issue by gilbertohasnofb** [Juanvvc/c172p-detailed#700]
Stuart wrote in the list:

> his looks good.  One bug that I think would be worth fixing before
> the release is that the DME display has been flipped upside down.
> 
> Do you want to fix that before I commit the new version?

I don't know what he means by flipped upside down, but the problem I see is that the display doesn't show anything. This was before:

![fgfs-screen-007](https://cloud.githubusercontent.com/assets/5700795/12931499/85698bf0-cf7e-11e5-8ff6-ee5bdebd779d.png)

And this is now:

![fgfs-screen-002](https://cloud.githubusercontent.com/assets/5700795/12931522/a1a3e0a4-cf7e-11e5-9714-26c1e1528639.png)

Also, the word MIN is flipped upside down, but note that KTS in the image above isn't.

![fgfs-screen-003](https://cloud.githubusercontent.com/assets/5700795/12931543/ba9abd62-cf7e-11e5-9f92-2f4869588aac.png)

**Comment by gilbertohasnofb** [comment]
I did not add a milestone as Stuart offered to postpone the merge so we can handle this. If that's the case, then @wlbragg could also mention jarlarntzen's name in the contributor's list. What do you all think, @onox, @wlbragg?

**Comment by wlbragg** [comment]
I'm OK with this. When I was re-texturing the DME several of the labels were backwards and upside down, I thought I accounted for all of them but I guess not. I also noticed the led digits didn't appear to be in place but didn't know if that was a bug or if it just didn't show until power was applied or time/mileage was achieved. I can work on it right now.

**Comment by gilbertohasnofb** [comment]
>. I can work on it right now.

Great, let me know if I can help with anything. Then I guess we just open a PR and wait for onox to see what to do.

**Comment by wlbragg** [comment]
I need help figuring out why the LED's are not displaying, if that is true?

**Comment by gilbertohasnofb** [comment]
It seems that the plane you applied the texture into is covering the display or something like this. Did you create that object or did you simply UV-mapped it?

**Comment by wlbragg** [comment]
I don't think so because the . (period) is showing, unless it is at a different depth than the rest of the digits.

**Comment by wlbragg** [comment]
> Did you create that object or did you simply UV-mapped it?

I borrows it from the AP and made sure it was in the background.

**Comment by gilbertohasnofb** [comment]
Does it display properly in Blender? Maybe just for the sake of troubleshooting you can try moving the background a lot sideways so you can see what's behind it in the sim. If everything displays properly then the 3D background object is the culprit, else it's something else.

**Comment by wlbragg** [comment]
I'll save it locally with the base plate (the plane) turned off and see if they show.

**Comment by gilbertohasnofb** [comment]
:+1:

**Comment by wlbragg** [comment]
Yes it is displaying correctly in Blender, let me run this test and see if it shows up without the base.

**Comment by wlbragg** [comment]
It still doesn't appear to show, their may be a name mismatch in the object VS XML. Where is the code that drives the digits?

**Comment by gilbertohasnofb** [comment]
`/Models/Interior/Panel/Instruments/dme/dme.xml`. or `ki266.xml`

**Comment by gilbertohasnofb** [comment]
By the way, even if the background is not the culprit, I think it needs a hole for the display, just like in the first picture I posted

**Comment by wlbragg** [comment]
They look OK, how does this device work, What should I see just powering it on, 00.0 000?

**Comment by wlbragg** [comment]
The first picture looks identical to the rest, no 0's showing, only the period?

**Comment by gilbertohasnofb** [comment]
I think it doesn't show anything unless you have tuned a station which transmits DME information

**Comment by gilbertohasnofb** [comment]
let me make a test here, see if I tune to a station it shows properly

**Comment by gilbertohasnofb** [comment]
It is working! 

![fgfs-screen-001](https://cloud.githubusercontent.com/assets/5700795/12933652/7ec588fc-cf89-11e5-9c4b-d9caed959f8f.png)

Sorry about the bogus report then, when I saw the display covered by the background I just assumed there was a problem... so the only things to be done here is to make a hole in the background and to invert that 
MIN tag.

**Comment by wlbragg** [comment]
>so the only things to be done here is to make a hole in the background 

I already inverted the MIN, but I don't understand "make a hole in the background"?

**Comment by gilbertohasnofb** [comment]
Look at the first image I posted in the report: the background is grey but the display is black, as it should be. Now the background covers both the grey area (correct) and the black display (incorrect).

**Comment by wlbragg** [comment]
I think I get what your talking about. For the digits and the MILES/MIN and KTS?

**Comment by wlbragg** [comment]
Unfortunately, that requires some cutting, it's going to take a few to do that. But first I want to make sure I only cut what is absolutely necessary.

**Comment by gilbertohasnofb** [comment]
This is the window display I mean:

![fg](https://cloud.githubusercontent.com/assets/5700795/12933834/a1e22196-cf8a-11e5-8145-7c7e9e6e9556.jpg)

**Comment by gilbertohasnofb** [comment]
Yes it will require some cutting

**Comment by wlbragg** [comment]
So we want the entire trapezoid (or whatever that shape is) cut out? I think that will show the lightgrey background then?

Without the new background
![dmea](https://cloud.githubusercontent.com/assets/11069680/12933976/e5eb14b8-cf50-11e5-85ae-5729fd6c52ad.jpg)

With the new background
![dmeb](https://cloud.githubusercontent.com/assets/11069680/12933986/f800e9ac-cf50-11e5-99d6-42a17dd4b13e.jpg)

**Comment by wlbragg** [comment]
I could try to give the trapezoid its own vertex (plane) at the backside and then that should block the new background, maybe?

**Comment by gilbertohasnofb** [comment]
> So we want the entire trapezoid (or whatever that shape is) cut out? 

Yes, that's what I am proposing

> I think that will show the lightgrey background then?

How come it isn't black? It should look like the image I posted, which was taken before we start modifying the DME.

**Comment by gilbertohasnofb** [comment]
> I could try to give the trapezoid its own vertex (plane) at the backside and then that should block the new background, maybe? 

I have no idea what you just wrote, but if it solves the problem then yaaay :stuck_out_tongue:

**Comment by gilbertohasnofb** [comment]
Just to be clear, this is the image I am talking about, note how the trapezoid display is black:

![](https://cloud.githubusercontent.com/assets/5700795/12931499/85698bf0-cf7e-11e5-8ff6-ee5bdebd779d.png)

**Comment by wlbragg** [comment]
>How come it isn't black?

I don't know, there was a very large plane in the dme.ac that I have turned off because it was of no use, unless it was causing the black background, I don't know if you know that this device is in two different ac's. The plane that I turned off was in the dme.ac which is the left on/off and the knob. The tight side is what we are talking about and it is all in the ki266.ac, this is where I added the new frame with the new background texture.

>Just to be clear

Yes, I understand.

Let me try something and see if I can get a black background in the trapezoid.

**Comment by wlbragg** [comment]
>here was a very large plane in the dme.ac that I have turned off because it was of no use, unless it was causing the black background,

Which I doubt because it would have caused the entire panel to be black.

**Comment by gilbertohasnofb** [comment]
Maybe what can be done is to create another 2D plane in the form of the trapezoid and leave it black. Then it would be in front of the metal background but behind the digits and other display information.

**Comment by gilbertohasnofb** [comment]
I mean, you know much more about 3D modelling than I do, I am just trying to help but I might be only speaking rubbish... so please ignore any stupid ideas :wink:

**Comment by wlbragg** [comment]
I'm basically doing what you suggest only using an existing object for the plane. I have it working in Blender, now to see it in the aircraft.
![dmec](https://cloud.githubusercontent.com/assets/11069680/12934351/823d1b8e-cf53-11e5-971f-f3de09c5fde1.jpg)

**Comment by gilbertohasnofb** [comment]
Ok, it's looking good. To test in the sim, you can use any VOR DME station, e.g. starting at LFSB and tuning to 117.45 in the NAV1

**Comment by wlbragg** [comment]
OK, let me try with the radio tuned

![dmed](https://cloud.githubusercontent.com/assets/11069680/12934459/51187ff2-cf54-11e5-9510-e7408ecf3837.jpg)

**Comment by wlbragg** [comment]
![dmef](https://cloud.githubusercontent.com/assets/11069680/12934517/b8669c98-cf54-11e5-88b5-385af232e52e.jpg)

We good?

**Comment by gilbertohasnofb** [comment]
Almost: why he frame around the trapezoid isn't light grey any more? Was that necessary for the black background?

**Comment by wlbragg** [comment]
Yeah I had to flip the normals, you want me to try to get it the other way?

**Comment by gilbertohasnofb** [comment]
Let me first see a picture of this DME in RL

**Comment by wlbragg** [comment]
OK, I'm still going to look at the normals a little closer while you do that.

**Comment by gilbertohasnofb** [comment]
![](http://i.ebayimg.com/00/s/NTA1WDczMg==/z/eOAAAOSwT6pVjqBo/$_1.JPG)

**Comment by wlbragg** [comment]
I know what I did wrong, hang on.

**Comment by wlbragg** [comment]
![dmeg](https://cloud.githubusercontent.com/assets/11069680/12934694/1efe65ac-cf56-11e5-821d-4cbce727caf7.jpg)

**Comment by gilbertohasnofb** [comment]
:clap:

**Comment by wlbragg** [comment]
This should have taken maybe 5 minutes. I added an edge face to the back of the trapezoid, but the normals pointed back into the dash. So either you flip them, which flips everything else, or you make it double sided, which then gave us what we wanted.

**Comment by gilbertohasnofb** [comment]
Well, it is solved now so that's great. Would you open a PR? I just want to make a quick test in the sim to make sure everything works as it should, and then I will also leave it for onox to merge it.

**Comment by wlbragg** [comment]
:+1:
